### PR TITLE
`EntityState` marker interface

### PR DESCRIPTION
**Pull request by dmitrykuzmin**
This PR adds a new marker interface for the messages representing entity state.

From now on, all messages marked as `(entity)` of some kind will be marked as `EntityState`. 

They can thus be distinguished from the generic messages and other message types such as `EventMessage` e.g. when creating method overloads.

The interface which is applied to all entity state messages can be configured as follows:

```gradle
modelCompiler {
    interfaces {
        mark messages().entityState(), asType("some.custom.EntityState")
    }
}
```

Please note, that just like for all other marker interface types, the specified custom interface should extend the `io.spine.base.EntityState` interface which is used by default.

Spine version advances to `1.2.0`.

### CODE REVIEW

**Comment by dmitrykuzmin** [comment]
@armiol PTAL.

**Comment by dmitrykuzmin** [comment]
@dmdashenkov PTAL.

File: tools/protoc-api/src/main/java/io/spine/tools/protoc/EntityState.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.protoc;
+
+/**
+ * Selects messages that represent an entity state.
+ */
+final class EntityState extends MessageSelector {
```
Comment (dmdashenkov) [inline tools/protoc-api/src/main/java/io/spine/tools/protoc/EntityState.java]: I don't understand how does this select entity states.

File: pom.xml [hunk 2]
```diff
@@ -12,7 +12,7 @@ all modules and does not describe the project structure per-subproject.
 
 <groupId>io.spine</groupId>
 <artifactId>spine-base</artifactId>
-<version>1.1.9</version>
+<version>1.1.11</version>
```
Comment (armiol) [inline pom.xml]: Please make sure to adjust the version before merging. There may be conflicts.

File: tools/protoc-api/src/main/java/io/spine/tools/protoc/EntityState.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.tools.protoc;
+
+/**
+ * Selects messages that represent an entity state.
```
Comment (armiol) [inline tools/protoc-api/src/main/java/io/spine/tools/protoc/EntityState.java]: This looks interesting combined with an "empty" implementation.

File: tools/protoc-api/src/main/java/io/spine/tools/protoc/GeneratedInterfaces.java [hunk 4]
```diff
@@ -116,6 +118,28 @@ public final void mark(UuidMessage uuidMessage, ClassName interfaceName) {
         uuidInterface = uuidConfig(interfaceName);
     }
 
+    /**
+     * Configures an interface generation for messages that represent an entity state.
+     *
+     * <p>All messages marked with {@code (entity)} option and with a valid {@code (kind)}
+     * will be marked with the given interface name.
+     *
+     * <p>Sample usage is:
```
Comment (armiol) [inline tools/protoc-api/src/main/java/io/spine/tools/protoc/GeneratedInterfaces.java]: `is as follows.` or just `Example:`.

File: tools/protoc-plugin/src/test/proto/spine/tools/protoc/iface/entity_states.proto [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+package spine.tools.protoc.iface;
+
+import "spine/options.proto";
+
+option (type_url_prefix) = "type.spine.io";
+option java_multiple_files = true;
+option java_outer_classname = "EntityStatesProto";
+option java_package = "io.spine.tools.protoc.iface";
+
+
```
Comment (dmdashenkov) [inline tools/protoc-plugin/src/test/proto/spine/tools/protoc/iface/entity_states.proto]: Please delete this empty line.

**Comment by dmitrykuzmin** [comment]
@dmdashenkov PTAL again.

**Pull request merged**